kernel/pipes.c

```diff
@@ -89,6 +102,8 @@ void z_impl_k_pipe_flush(struct k_pipe *pipe)
 	(void) pipe_get_internal(key, pipe, NULL, (size_t) -1, &bytes_read, 0U,
 				 K_NO_WAIT);
 
+	handle_poll_events(pipe);
+
```

Comment:
I don't think that calling this here is correct. Flush will empty the pipe. There will be no data to available to read after calling this.